great_expectations/expectations/model_field_types.py

```
@@ -22,11 +22,13 @@
 ]
 
 ValueSetField = Annotated[
-    Union[Sequence, set, SuiteParameterDict],
-    pydantic.Field(
+    Union[List, SuiteParameterDict],
+    Field(
         title="Value Set",
         description=VALUE_SET_DESCRIPTION,
+        # Add type constraint to exclude strings and bytes
         schema_overrides={
+            "not": {"anyOf": [{"type": "string"}, {"type": "string", "format": "binary"}]},
```

for reviewers - this prevents string and bytestring from coercing to a list. Should we skip this change and instead have an expectation validation error if someone passes a string instead of expected iterables (which is unlikely but possible)?
I am also not sure if we want to update schemas - can remove this addition and undo schema changes

---

Does this actually prevent the type coercion when using the API, or does it just change the schema? If it just changes the schema, I don't think it's necessary. You could raise an error in the validator if the wrong type is passed. This could be a more widespread problem so I wouldn't spend too much time thinking about it.

---

By "prevent from coercing" I meant it would prevent processing the validation since this schema update would raise a validation error:
```
pydantic.v1.error_wrappers.ValidationError: 2 validation errors for ExpectColumnValuesToBeInSet
value_set
  value is not a valid list (type=type_error.list)
value_set
  value is not a valid dict (type=type_error.dict)
```

---

Actually, I think we may not need it - I just tested without the schema changes locally and it still throws this validation error. I think it was coersing the string when we had `Sequence` which would allow string and bytes, but with `List` it's no longer the case. Going to undo this change!

---

I guess that's what my question was. I wouldn't expect it to raise a validation error if you are just updating the schema. Are you sure if you remove the schema changes that it will fail to raise the error?

---

This is now reverted - updated the diff